contrib/win32/install/install/product.wxs

```diff
@@ -0,0 +1,170 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<?if $(var.MANUFACTURER) = "" ?>
+	<?define MANUFACTURER = "Microsoft Corporation"?>
+<?endif?>
+
+<?if $(var.SOURCEDIR) = "" ?>
+  <?error Variable SOURCEDIR not set?>
+<?endif?>
+
+<?if $(var.VERSION) = "" ?>
+  <?define VERSION = "0.0.0.0"?>
+<?endif?>
+
+<?if $(sys.BUILDARCH)="x86" ?>
+  <?define PROGRAMFILESFOLDER="ProgramFilesFolder"?>
+<?elseif $(sys.BUILDARCH)="x64" ?>
+  <?define PROGRAMFILESFOLDER="ProgramFiles64Folder"?>
+<?else?>
+  <?error Unsupported value of sys.BUILDARCH=$(sys.BUILDARCH)?>
+<?endif?>
+
+<?define UPGRADECODERELEASE = "c7a00fb1-c477-40d3-a951-ec4c749da439" ?>
+<?define UPGRADECODEDEBUG = "e2dd3c95-7a8b-444e-b75b-3e1cf697aadc" ?>
+
+<?if $(var.PRODUCTID) = "" ?>
+  <?define PRODUCTID = "*" ?>
+<?endif?>
+<?if $(var.CONFIGURATION) ~= "Release" ?>
+  <?define UPGRADECODE = "$(var.UPGRADECODERELEASE)" ?>
+  <?define UPGRADECODEBLOCKED = "$(var.UPGRADECODEDEBUG)" ?>
+<?elseif $(var.CONFIGURATION) ~= "Debug" ?>
+  <?define UPGRADECODE = "$(var.UPGRADECODEDEBUG)" ?>
+  <?define UPGRADECODEBLOCKED = "$(var.UPGRADECODERELEASE)" ?>
+<?else?>
+  <?error Variable CONFIGURATION is not set or uses an unsupported value. Valid values: Debug, Release ?>
+<?endif?>
+
+<Wix
+  xmlns="http://schemas.microsoft.com/wix/2006/wi"
+  xmlns:util="http://schemas.microsoft.com/wix/UtilExtension"
+  xmlns:fw="http://schemas.microsoft.com/wix/FirewallExtension">
+	<Product
+    Id="$(var.PRODUCTID)"
+    Name="OpenSSH ($(var.CONFIGURATION) $(var.VERSION))"
+    Language="1033"
+    Version="$(var.VERSION)"
+    Manufacturer="$(var.MANUFACTURER)"
+    UpgradeCode="$(var.UPGRADECODE)">
+
+		<Package
+      InstallerVersion="200"
+      Compressed="yes"
+      InstallScope="perMachine"
+      InstallPrivileges="elevated"
+      />
+
+    <Property Id="LOGVERBOSE" Value="LOGVERBOSE"/>
+
+    <!-- https://stackoverflow.com/a/19664484 -->
+    <Upgrade Id="$(var.UPGRADECODEBLOCKED)">
```

Comment:
This wouldn't be necessary if properly authored as a major upgrade.

Reply:
I added this here because I wanted to prevent people from installing a debug version over a release version with the same product version string. But if this not required at all, or can be achieved by using different product code or by other means this can be removed for sure.

Reply:
It just makes the UX harder. If they want to flip between retail and debug, that should be up to them.

Reply:
Well okay, I've to admit maybe I really overshot the mark. So I will remove that part here 👍🏼

Reply:
Removed!